tests/upgrades/test_foreman_maintain.py

```diff
@@ -26,37 +26,85 @@ class ScenarioForemanMaintain(TestCase):
     Test Steps:
 
     1. Before Satellite upgrade, Perform test for
-       foreman-maintain list-versions.
+       foreman-maintain upgrade list-versions.
     2. Upgrade satellite/capsule.
-    3. Perform tests for foreman-maintain list-versions, after upgrade.
+    3. Perform tests for foreman-maintain upgrade list-versions, after upgrade.
     4. Check if tests passed.
 
     """
 
+    @staticmethod
+    def satellite_upgradable_version_list():
+        """
+        This function is used to collect the details of satellite version  and upgradable
+        version list.
+
+        :return: satellite_version, upgradeable_version
+        """
+
+        cmd = "rpm -q satellite > /dev/null && rpm -q satellite --queryformat=%{VERSION}"
+        satellite_version = ssh.command(cmd)
+        if not satellite_version.return_code:
+            satellite_version = satellite_version.stdout[0]
+        forman_maintain_version = ssh.command("foreman-maintain upgrade list-versions")
+        upgradeable_version = [version for version in forman_maintain_version.stdout
+                               if version != '']
+        return satellite_version, upgradeable_version
+
+    @staticmethod
+    def version_details(satellite_version, upgrade_stage="pre-upgrade"):
+        """
+        This function is used to update the details of zstream upgrade and
+        next version upgrade
+        :param str satellite_version: satellite version would be like 6.5.0, 6.6.0, 6,7.0
+        :param str upgrade_stage: upgrade stage would be pre or post.
+        :return: zstream_version, next_version
+        """
+
+        major_version = satellite_version.split('.')[0:1]
+        y_version = int(satellite_version.split('.')[0:2][-1])
+        zstream_version = ''
+        if upgrade_stage == "pre-upgrade":
+            major_version.append(str(y_version + 1))
```

Comment:
This will fail in the scenario i.e.    Satellite 6.x ends at 6.8.0  and the next version starts from Satellite 7.x , right ?

Reply:
Fixed.